### PR TITLE
Feature/gb3 1390 info with image preview

### PR DESCRIPTION
**Pull request by TIL-EBP**
Adds images to the feature info. 
Links to working examples: 
http://localhost:4200/s/7101c634-b835-47ea-98e3-7951257c03a3
http://localhost:4200/s/ce842c54-382a-4720-bfed-6373c1f8328a (need to be looged in)
In some cases, the src they send is just a link to a hosting platform an thus cannot display an image. They are working on it (see [Ticket](https://jira-geo.zh.ch/browse/GB3-1390)

### CODE REVIEW

File: src/app/map/components/feature-info-overlay/feature-info-content/feature-info-content.component.html [hunk 1]
```diff
@@ -37,18 +37,34 @@
...
+          @switch (cellValue.cellType) {
+            @case ('url') {
+              <a
+                class="feature-info-content__table__row__column__link"
+                [href]="cellValue.url"
+                target="_blank"
+                rel="noopener noreferrer"
+                [title]="cellValue.displayValue"
+              >
+                {{ cellValue.displayValue
```
Comment (Tugark) [inline src/app/map/components/feature-info-overlay/feature-info-content/feature-info-content.component.html:49]: Weird linebreak?

File: src/app/shared/interfaces/feature-info.interface.ts [hunk 2]
```diff
@@ -2,10 +2,11 @@ import {GeometryWithSrs} from './geojson-types-with-srs.interface';
 import {HasMetaDataLink} from './metaDataLink.interface';
 import {IsSingleLayer} from './single-layer.interface';
 import {LinkObject} from './link-object.interface';
+import {Image} from '../models/gb3-api-generated.interfaces';
```
Comment (Tugark) [inline src/app/shared/interfaces/feature-info.interface.ts:5]: I add this comment here, but it also holds true for others: With this, your essentially coupling all internal components and interfaces to the API. If they change it, our interfaces and our components will break.

Generated interfaces should be abstracted and only used within the services, and then mapped.

File: src/app/shared/services/apis/gb3/gb3-topics.service.ts [hunk 3]
```diff
@@ -369,14 +370,20 @@ export class Gb3TopicsService extends Gb3ApiService {
...
-  private createFeatureInfoFieldValue(field: InfoFeatureField): string | LinkObject | null {
-    if ('link' in field) {
+  private createFeatureInfoFieldValue(field: InfoFeatureField): string | LinkObject | Image | null {
+    if (field.value === null) {
+      return null;
+    }
+    if (typeof field.value === 'number') {
+      return field.value.toString();
+    }
+    if (typeof field.value === 'string' || 'alt' in field.value) {
```
Comment (Tugark) [inline src/app/shared/services/apis/gb3/gb3-topics.service.ts:380]: That one is quite implicit. Maybe add a comment explaining that you check for string OR image, and if both fail, it must be a URL.

**Review by Tugark — CHANGES_REQUESTED** [review]
In general, it looks okay.

However, the interfaces are not correct. We specified in https://jira-geo.zh.ch/browse/GB3-1620 that we need the types in the interface, because this would allow us to use a switch case as per PSI's last comment. Your "hacky" solution for figuring out the actual type of the field works, but it's quite hard to read.

I also think that at one point, the union type was specified, since we have the same for the VectorLayers. Maybe we can discuss this.

File: src/app/shared/interfaces/feature-info.interface.ts (hunk 2 above)
Comment (TIL-EBP) [reply src/app/shared/interfaces/feature-info.interface.ts:5]: I created a new interface which copies the generated Image interace (analog to LinkObject) and replaced all the imports

**Review by Tugark — APPROVED** [review]

**Review by Tugark — APPROVED** [review]

**Pull request merged**